infra/messaging/src/index.js

```diff
@@ -123,80 +126,330 @@ app.post('/accounts/:address', async (req, res) => {
   const { signature, data } = req.body
 
   if (
-    verifyRegistrySignature(signature, '', {
-      payload: { value: data, key: address }
+    verifyRegistrySignature(signature, {
+      payload: {
+        value: data,
+        key: address
+      }
     })
   ) {
     const entry = await db.Registry.findOne({ where: { ethAddress: address } })
     console.log('setting registry existing entry:', entry)
-    if (!entry || entry.signature != signature) {
+    if (!entry || entry.signature !== signature) {
       await db.Registry.upsert({ ethAddress: address, data, signature })
     }
+
     return res.status(200).send(address)
   }
   res.statusMessage = 'Cannot verify signature of registry'
   return res.status(400).end()
 })
 
+// Fetch total unread message count of an address
+app.get('/conversations/:address/unread', async (req, res) => {
+  let { address } = req.params
+
+  if (!Web3Utils.isAddress(address)) {
+    res.statusMessage = `Address '${address}' is not a valid Ethereum address`
+
+    return res.status(400).end()
+  }
+
+  address = Web3Utils.toChecksumAddress(address)
+
+  // TODO: Get rid of the custom query and find the correct ORM way
+
+  try {
+    const [[convs]] = await db.sequelize.query(
+      `
+SELECT SUM(unread) as unread FROM (SELECT messages.unread_count::integer as unread FROM msg_conversee conversee
+  LEFT JOIN 
+    (SELECT m.conversation_id, count(m.conversation_id)::integer as unread_count 
+      FROM msg_message m 
+      WHERE m.read=False AND m.is_keys=False
+        AND m.eth_address<>:address
+      GROUP BY m.conversation_id) messages
+    ON messages.conversation_id=conversee.conversation_id
+  WHERE conversee.eth_address=:address) unread_by_conv`,
+      {
+        replacements: {
+          address
+        }
+      }
+    )
+
+    return res.status(200).send(convs)
+  } catch (e) {
+    return res.status(500).send(e)
```

Comment:
Let's also log using logger.error
Same comment for all functions below so that we get visibility into errors when looking at the server logs.

Reply:
Have added logs to functions where we catch the errors. However, error handling is not in place for a lot of function and the response that messaging server returns is inconsistent (sometimes JSON and sometimes simply the text). I will be changing that in subsequent PRs.